pkgs/by-name/ra/ratchet/package.nix

```
@@ -0,0 +1,57 @@
+{
+  lib,
+  buildGoModule,
+  fetchFromGitHub,
+  callPackage,
+}:
+buildGoModule rec {
+  pname = "ratchet";
+  version = "0.9.2";
+
+  src = fetchFromGitHub {
+    owner = "sethvargo";
+    repo = "ratchet";
+    rev = "d57cc1a53c022d3f87c4820bc6b64384a06c8a07";
+    hash = "sha256-gQ98uD9oPUsECsduv/lqGdYNmtHetU49ETfWCE8ft8U=";
+  };
+
+  proxyVendor = true;
+  vendorHash = "sha256-J7LijbhpKDIfTcQMgk2x5FVaYG7Kgkba/1aSTmgs5yw=";
+
+  subPackages = ["."];
+
+  ldflags = let
+    package_url = "github.com/sethvargo/ratchet";
+  in [
+    "-s"
+    "-w"
+    "-X ${package_url}/internal/version.name=${pname}"
+    "-X ${package_url}/internal/version.version=${version}"
+    "-X ${package_url}/internal/version.commit=${src.rev}"
+  ];
+
+  doInstallCheck = true;
+  installCheckPhase = ''
+    $out/bin/ratchet --version 2>&1 | grep ${version};
+  '';
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/bin
+    install -Dm755 "$GOPATH/bin/ratchet" -T $out/bin/ratchet
+    runHook postInstall
+  '';
+
+  passthru.tests = {
+    execution = callPackage ./tests.nix {};
+  };
+
+  meta = with lib; {
+    description = "Ratchet: A tool for securing CI/CD workflows with version pinning.";
+    mainProgram = "ratchet";
+    downloadPage = "https://github.com/sethvargo/ratchet";
+    homepage = "https://github.com/sethvargo/ratchet";
+    license = licenses.asl20;
+    maintainers = with maintainers; [cameronraysmith];
```

```suggestion
    maintainers = with maintainers; [ cameronraysmith ];
```

You may want to run `nix shell nixpkgs/master#nixfmt-rfc-style -c nixfmt pkgs/by-name/ra/ratchet/package.nix` to auto format it to match conventions

---

Thank you for this pointer. I previously formatted with alejandra. I'll update with the recommended formatter.